Remove the need for freetype if not needed

On a regular default ubuntu 16 using AF master (as today) :
$cmakebin ..
-DAF_WITH_GRAPHICS=OFF
-DAF_WITH_STATIC_FREEIMAGE=OFF
-DAF_WITH_IMAGEIO=OFF
...
CMake Error at /home/wtambellini/repos/xmt-externals/FC12/tools/cmake/share/cmake-3.9/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
Could NOT find Freetype (missing: FREETYPE_LIBRARY) (found version "2.6.1")
Call Stack (most recent call first):
/home/wtambellini/repos/xmt-externals/FC12/tools/cmake/share/cmake-3.9/Modules/FindPackageHandleStandardArgs.cmake:377 (_FPHSA_FAILURE_MESSAGE)
/home/wtambellini/repos/xmt-externals/FC12/tools/cmake/share/cmake-3.9/Modules/FindFreetype.cmake:140 (find_package_handle_standard_args)
extern/forge/CMakeLists.txt:48 (find_package)

That machine does seem to have the freetype dev :
$ locate libfreetype.
/usr/lib/i386-linux-gnu/libfreetype.so.6
/usr/lib/i386-linux-gnu/libfreetype.so.6.12.1
/usr/lib/x86_64-linux-gnu/libfreetype.a
/usr/lib/x86_64-linux-gnu/libfreetype.la
/usr/lib/x86_64-linux-gnu/libfreetype.so
/usr/lib/x86_64-linux-gnu/libfreetype.so.6
/usr/lib/x86_64-linux-gnu/libfreetype.so.6.12.1

$ sudo apt-get install libfreetype6-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libfreetype6-dev is already the newest version (2.6.1-0.1ubuntu2.3).

Should nt we be able to build a non graphics AF without the need for freetype dev if the user does nt need to do any GUI things?

--- a/CMakeModules/AFconfigure_forge_submodule.cmake
+++ b/CMakeModules/AFconfigure_forge_submodule.cmake
@@ -5,35 +5,43 @@
 # The complete license agreement can be obtained at:
 # http://arrayfire.com/licenses/BSD-3-Clause
 
-set(ArrayFireInstallPrefix ${CMAKE_INSTALL_PREFIX})
-set(ArrayFireBuildType ${CMAKE_BUILD_TYPE})
-set(CMAKE_INSTALL_PREFIX ${ArrayFire_BINARY_DIR}/extern/forge/package)
-set(CMAKE_BUILD_TYPE Release)
-set(FG_BUILD_EXAMPLES OFF CACHE BOOL "Used to build Forge examples")
-set(FG_BUILD_DOCS OFF CACHE BOOL "Used to build Forge documentation")
-set(FG_WITH_FREEIMAGE OFF CACHE BOOL "Turn on usage of freeimage dependency")
-if (AF_BUILD_FORGE)
-  add_subdirectory(extern/forge)
-else (AF_BUILD_FORGE)
-  add_subdirectory(extern/forge EXCLUDE_FROM_ALL)
-endif (AF_BUILD_FORGE)
-mark_as_advanced(
-    FG_BUILD_EXAMPLES
-    FG_BUILD_DOCS
-    FG_WITH_FREEIMAGE
-    FG_USE_WINDOW_TOOLKIT
-    FG_USE_SYSTEM_CL2HPP
-    FG_ENABLE_HUNTER
-    glfw3_DIR
-    glm_DIR
-    )
-set(CMAKE_BUILD_TYPE ${ArrayFireBuildType})
-set(CMAKE_INSTALL_PREFIX ${ArrayFireInstallPrefix})
+if(AF_BUILD_FORGE)
+  set(ArrayFireInstallPrefix ${CMAKE_INSTALL_PREFIX})
+  set(ArrayFireBuildType ${CMAKE_BUILD_TYPE})
+  set(CMAKE_INSTALL_PREFIX ${ArrayFire_BINARY_DIR}/extern/forge/package)
+  set(CMAKE_BUILD_TYPE Release)
+  set(FG_BUILD_EXAMPLES OFF CACHE BOOL "Used to build Forge examples")
+  set(FG_BUILD_DOCS OFF CACHE BOOL "Used to build Forge documentation")
+  set(FG_WITH_FREEIMAGE OFF CACHE BOOL "Turn on usage of freeimage dependency")
 
-if (AF_BUILD_FORGE)
+  add_subdirectory(extern/forge)
+
+  mark_as_advanced(
+      FG_BUILD_EXAMPLES
+      FG_BUILD_DOCS
+      FG_WITH_FREEIMAGE
+      FG_USE_WINDOW_TOOLKIT
+      FG_USE_SYSTEM_CL2HPP
+      FG_ENABLE_HUNTER
+      glfw3_DIR
+      glm_DIR
+      )
+  set(CMAKE_BUILD_TYPE ${ArrayFireBuildType})
+  set(CMAKE_INSTALL_PREFIX ${ArrayFireInstallPrefix})
+
   install(FILES
       $<TARGET_FILE:forge>
       $<TARGET_SONAME_FILE:forge>
       DESTINATION "${AF_INSTALL_LIB_DIR}"
       COMPONENT common_backend_dependencies)
-endif (AF_BUILD_FORGE)
+else(AF_BUILD_FORGE)
+  set(FG_VERSION "1.0.0")
+  set(FG_VERSION_MAJOR 1)
+  set(FG_VERSION_MINOR 0)
+  set(FG_VERSION_PATCH 0)
+  set(FG_API_VERSION_CURRENT 10)
+  configure_file(
+    ${PROJECT_SOURCE_DIR}/extern/forge/CMakeModules/version.h.in
+    ${PROJECT_BINARY_DIR}/extern/forge/include/fg/version.h
+    )
+endif(AF_BUILD_FORGE)

